### ISSUE TITLE
Ironwail not launching

### ISSUE THREAD

**Issue by vaibhavrmahadevan** [luxtorpeda-dev/packages#756]
### Game

Quake

### Engine

Ironwail

### Linux Distribution

Linux Mint 21

### Luxtorpeda Version

59

### Bug description

Ironwail does not launch from Steam.

### Steps To Reproduce

1. Launching Quake
3. Picking Ironwail from the luxtorpeda menu

### Expected Behavior

Ironwail launches in a new window

### Relevant log output

```shell
GameAction [AppID 2310, ActionID 1] : LaunchApp changed task to ShowLaunchOption with ""
GameAction [AppID 2310, ActionID 1] : LaunchApp waiting for user response to ShowLaunchOption ""
Could not connect to X session manager: None of the authentication protocols specified are supported
Could not connect to X session manager: None of the authentication protocols specified are supported
GameAction [AppID 2310, ActionID 1] : LaunchApp continues with user response "0"
GameAction [AppID 2310, ActionID 1] : LaunchApp changed task to ProcessingInstallScript with ""
14:24:46 [INFO] setup_logging success
Error: Custom { kind: Other, error: "iscriptevaluator ignoring" }
GameAction [AppID 2310, ActionID 1] : LaunchApp changed task to SynchronizingCloud with ""
GameAction [AppID 2310, ActionID 1] : LaunchApp changed task to SynchronizingControllerConfig with ""
GameAction [AppID 2310, ActionID 1] : LaunchApp changed task to SiteLicenseSeatCheckout with ""
GameAction [AppID 2310, ActionID 1] : LaunchApp changed task to CreatingProcess with ""
GameAction [AppID 2310, ActionID 1] : LaunchApp waiting for user response to CreatingProcess ""
GameAction [AppID 2310, ActionID 1] : LaunchApp continues with user response "CreatingProcess"
/bin/sh\0-c\0/home/vaibhavrm/.steam/debian-installation/ubuntu12_32/reaper SteamLaunch AppId=2310 -- /home/vaibhavrm/.steam/debian-installation/ubuntu12_32/steam-launch-wrapper -- '/media/Steam/SteamLibrary/steamapps/common/SteamLinuxRuntime_soldier'/_v2-entry-point --verb=waitforexitandrun -- '/home/vaibhavrm/.steam/debian-installation/compatibilitytools.d/luxtorpeda'/luxtorpeda.sh waitforexitandrun  '/media/Steam/SteamLibrary/steamapps/common/Quake/rerelease/Quake_x64_steam.exe'\0
Game process added : AppID 2310 "/home/vaibhavrm/.steam/debian-installation/ubuntu12_32/reaper SteamLaunch AppId=2310 -- /home/vaibhavrm/.steam/debian-installation/ubuntu12_32/steam-launch-wrapper -- '/media/Steam/SteamLibrary/steamapps/common/SteamLinuxRuntime_soldier'/_v2-entry-point --verb=waitforexitandrun -- '/home/vaibhavrm/.steam/debian-installation/compatibilitytools.d/luxtorpeda'/luxtorpeda.sh waitforexitandrun  '/media/Steam/SteamLibrary/steamapps/common/Quake/rerelease/Quake_x64_steam.exe'", ProcID 44342, IP 0.0.0.0:0
chdir /media/Steam/SteamLibrary/steamapps/common/Quake
ERROR: ld.so: object '/home/vaibhavrm/.steam/debian-installation/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
GameAction [AppID 2310, ActionID 1] : LaunchApp changed task to WaitingGameWindow with ""
ERROR: ld.so: object '/home/vaibhavrm/.steam/debian-installation/ubuntu12_64/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS64): ignored.
GameAction [AppID 2310, ActionID 1] : LaunchApp changed task to Completed with ""
ERROR: ld.so: object '/home/vaibhavrm/.steam/debian-installation/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
ERROR: ld.so: object '/home/vaibhavrm/.steam/debian-installation/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
ERROR: ld.so: object '/home/vaibhavrm/.steam/debian-installation/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
ERROR: ld.so: object '/home/vaibhavrm/.steam/debian-installation/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
pid 44350 != 44346, skipping destruction (fork without exec?)
Installing breakpad exception handler for appid(steam)/version(1664418568)
LD_PRELOAD found with /tmp/pressure-vessel-libs-CY2KT1/${PLATFORM}/gameoverlayrenderer.so
pid 44546 != 44545, skipping destruction (fork without exec?)
14:24:50 [INFO] setup_logging success
14:24:50 [INFO] turning virtual gamepad off
14:24:50 [INFO] luxtorpeda version: 59.0.0
14:24:50 [INFO] steam_app_id: "2310"
14:24:50 [INFO] original command: ["/media/Steam/SteamLibrary/steamapps/common/Quake/rerelease/Quake_x64_steam.exe"]
14:24:50 [INFO] working dir: Ok("/media/Steam/SteamLibrary/steamapps/common/Quake")
14:24:50 [INFO] tool dir: "/home/vaibhavrm/.steam/debian-installation/compatibilitytools.d/luxtorpeda"
14:24:50 [INFO] user_packages_file not found
14:24:50 [INFO] user_packages_file not found
14:24:50 [INFO] showing engine choices
14:24:50 [INFO] window is on display_index: 0 on_steam_deck: false steam_deck_gaming_mode: false
14:24:50 [INFO] found dpi: (141.67935, 141.76744, 141.40207) using dpi: 141.76744
14:24:50 [INFO] using scaled_width: 705 scaled_height: 470
14:24:50 [INFO] using dpi scaling of 1.8459301
14:24:50 [INFO] 1 joysticks available
14:24:50 [INFO] Attempting to open controller 0
14:24:50 [INFO] controller name is X360 Controller
14:24:50 [INFO] Success: opened "X360 Controller"
14:24:50 [INFO] Controller connected mapping: 030000005e0400008e02000010010000,X360 Controller,a:b0,b:b1,back:b6,dpdown:h0.4,dpleft:h0.8,dpright:h0.2,dpup:h0.1,guide:b8,leftshoulder:b4,leftstick:b9,lefttrigger:a2,leftx:a0,lefty:a1,rightshoulder:b5,rightstick:b10,righttrigger:a5,rightx:a3,righty:a4,start:b7,x:b2,y:b3,platform:Linux
14:24:50 [INFO] controller assumed to be xbox
using dpi_value: 141.76744
Installing breakpad exception handler for appid(steam)/version(1664418568)
14:24:54 [INFO] engine choice: "ironwail"
14:24:54 [INFO] engine choice complete
14:24:54 [INFO] ironwail-common-3.tar.xz found in cache (skip)
14:24:54 [INFO] engine choice complete
14:24:54 [INFO] json:
14:24:54 [INFO] {
    "game_name": "Quake",
    "download": [
        {
            "name": "ironwail",
            "url": "https://github.com/luxtorpeda-dev/packages/releases/download/ironwail-3/",
            "file": "ironwail-common-3.tar.xz",
            "cache_by_name": true
        }
    ],
    "cloudAvailable": true,
    "controllerSteamDefault": true,
    "command": "./ironwail/ironwail.sh",
    "engine_name": "ironwail"
}
14:24:54 [INFO] installing: ironwail-common-3
14:24:54 [INFO] install: "ironwail"
14:24:54 [INFO] install: "ironwail/ironwail"
14:24:54 [INFO] install: "ironwail/ironwail.sh"
14:24:54 [INFO] install: "ironwail/share"
14:24:54 [INFO] install: "ironwail/share/quake"
14:24:54 [INFO] install: "ironwail/share/quake/id1"
14:24:54 [INFO] install: "ironwail/share/quake/id1/pak0.pak"
14:24:54 [INFO] install: "ironwail/share/quake/id1/pak1.pak"
14:24:54 [INFO] install: "ironwail/share/quake/default.lux.cfg"
14:24:54 [INFO] install: "ironwail/share/quake/hipnotic"
14:24:54 [INFO] install: "ironwail/share/quake/hipnotic/pak0.pak"
14:24:54 [INFO] install: "ironwail/share/quake/mg1"
14:24:54 [INFO] install: "ironwail/share/quake/mg1/pak0.pak"
14:24:54 [INFO] install: "ironwail/share/quake/rogue"
14:24:54 [INFO] install: "ironwail/share/quake/rogue/pak0.pak"
14:24:54 [INFO] install: "ironwail/share/quake/dopa"
14:24:54 [INFO] install: "license"
14:24:54 [INFO] install: "license/LICENSE.ironwail"
14:24:54 [INFO] install: "license/mad.COPYRIGHT"
14:24:54 [INFO] install: "license/..LICENSE"
14:24:54 [INFO] install: "license/mad.COPYING"
14:24:54 [INFO] install: "lib"
14:24:54 [INFO] install: "lib/libmad.so.0"
14:24:54 [INFO] install: "lib/libmad.so"
14:24:54 [INFO] install: "lib/libmad.so.0.2.1"
14:24:54 [INFO] run: "./ironwail/ironwail.sh" with args: [] []
Command line: ./ironwail/ironwail -basedir ironwail/share/quake
Found SDL version 2.24.0

ERROR-OUT BEGIN


QUAKE ERROR: Your version of SDL library is incompatible with me.
You need a library version in the line of 2.0.0


14:24:54 [INFO] run returned with exit status: 1
```

**Comment by d10sfan** [comment]
@vaibhavrmahadevan Thanks for the report! It looks like this is related to steamrt updating to sdl2 2.24, which would defeat that library version check.

Looks like this happens here: https://github.com/andrei-drexler/ironwail/blob/eeec028d767542905bcfacb09fcffd836d072d6a/Quake/main_sdl.c#L33

I should be able to make a patch to allow it to work with 2.24, and will test to see if anything goes wrong. I also updated the steamrt used for building to the latest.

**Comment by vaibhavrmahadevan** [comment]
@d10sfan Thank you so much! I had a feeling quakespasm-spiked might also have been affected by the same issue :)

**Comment by d10sfan** [comment]
@vaibhavrmahadevan Thanks again for reporting! 

Fixed with the following commits:

https://github.com/luxtorpeda-dev/packages/commit/0d5d290f6c718a2fcd5917c54d53e4b43271c6ea (anything after this will be built with the new steamrt)

https://github.com/luxtorpeda-dev/packages/commit/708b65dd978850644bc31c978fb37574abe801ea
https://github.com/luxtorpeda-dev/packages/commit/ad9670d28b4f3e4bcacb42f91940cbdb6d8380ed
https://github.com/luxtorpeda-dev/packages/commit/991c0f81c35a9e9ec413fb639ee64b21afb71bf2

ironwail and quakespasm-spiked should work now. Both of them had changes that removed that check (for the same reason this occurred it looks like) so I updated to those.

Let me know if you run into any issues.
